#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_glpk-pre-ifdef-logic.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## DP: Fixed #ifdef logic around macro GLPK_PRE_4_14, which was causing 
## DP: an extra closing brace to be introduced for the extern "C" 
## DP: declaration when GLPK_PRE_4_14 was not defined.  This patch has been 
## DP: sent to bug@octave.org.
 

@DPATCH@

--- octave2.9-2.9.10.orig/src/DLD-FUNCTIONS/__glpk__.cc
+++ octave2.9-2.9.10/src/DLD-FUNCTIONS/__glpk__.cc
@@ -53,17 +53,14 @@
 #ifndef lib_set_print_hook
 #define lib_set_print_hook lib_print_hook
 #endif
-}
 
 #else
 
-extern "C"
-{
 void _glp_lib_print_hook (int (*func)(void *info, char *buf), void *info);
 void _glp_lib_fault_hook (int (*func)(void *info, char *buf), void *info);
-}
 
 #endif
+}
 
 #define NIntP 17
 #define NRealP 10
